### PR TITLE
test(block, block-section): add component token E2E tests

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #7180

## Summary

✨🧪✨

### CODE REVIEW

File: packages/calcite-components/src/components/block-section/block-section.e2e.ts [hunk 1]
```diff
@@ -232,4 +232,97 @@ describe("calcite-block-section", () => {
...
+        "--calcite-block-section-switch-handle-border-color-hover": {
+          shadowSelector: `calcite-switch`,
+          targetProp: "--calcite-switch-handle-border-color",
+          state: "hover",
+        },
+        "--calcite-block-section-switch-track-background-color": {
+          shadowSelector: `calcite-switch`,
+          targetProp: "--calcite-switch-track-background-color",
+        },
+        "--calcite-block-section-switch-track-border-color": {
```
Comment (alisonailea) [inline packages/calcite-components/src/components/block-section/block-section.e2e.ts:321]: this variable isn't being applied.

**Review by alisonailea — CHANGES_REQUESTED** [review]
Can you recheck these tests in debugger mode?

File: packages/calcite-components/src/components/block-section/block-section.e2e.ts [hunk 2]
```diff
@@ -232,4 +232,97 @@ describe("calcite-block-section", () => {
...
+          shadowSelector: `.${CSS.sectionHeader}`,
+          targetProp: "color",
+          state: "hover",
+        },
+      });
+    });
+
+    describe("status icon", () => {
+      themed(`<calcite-block-section status="valid"></calcite-block-section>`, {
+        "--calcite-block-section-status-icon-color": {
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply packages/calcite-components/src/components/block-section/block-section.e2e.ts:271]: > this is being overwritten by .status-icon.valid

This is not accurate. Internal, fallback, values are assigned when valid/invalid, but `--calcite-block-section-status-icon-color` has higher precedence. 

Screenshot 2024-06-26 at 3 06 10 PM



LMK if I'm missing something.

File: packages/calcite-components/src/components/block-section/block-section.e2e.ts [hunk 3]
```diff
@@ -232,4 +232,97 @@ describe("calcite-block-section", () => {
     expect(await element.getProperty("open")).toBe(false);
     expect(toggle.getAttribute("aria-expanded")).toBe("false");
   }
+
+  describe("theme", () => {
+    describe("default", () => {
+      themed(html`<calcite-block-section heading="heading" description="description"></calcite-block-section>`, {
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply packages/calcite-components/src/components/block-section/block-section.e2e.ts]: Fixed!

File: packages/calcite-components/src/components/block-section/block-section.e2e.ts [hunk 4]
```diff
@@ -232,4 +232,97 @@ describe("calcite-block-section", () => {
...
+        },
+      });
+    });
+
+    describe("open", () => {
+      themed(`<calcite-block-section open></calcite-block-section>`, {
+        "--calcite-block-section-border-color": {
+          targetProp: "borderColor",
+        },
+        "--calcite-block-section-heading-text-color-hover": {
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply packages/calcite-components/src/components/block-section/block-section.e2e.ts]: Fixed!

File: packages/calcite-components/src/components/block-section/block-section.e2e.ts (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/block-section/block-section.e2e.ts]: Fixed!

File: packages/calcite-components/src/components/block-section/block-section.scss [hunk 5]
```diff
@@ -7,9 +7,11 @@
  * @prop --calcite-block-section-border-color: Specifies the border color of the component. Applicable on bottom border when open.
  * @prop --calcite-block-section-heading-text-color-hover: Specifies the component's heading text color on hover.
  * @prop --calcite-block-section-heading-text-color: Specifies the component's heading text color.
+ * @prop --calcite-block-section-icon-color: Specifies the component's heading text color.
  * @prop --calcite-block-section-status-icon-color: Specifies the status icon color of the component.
  * @prop --calcite-block-section-switch-corner-radius: Specifies the border radius of the switch.
  * @prop --calcite-block-section-switch-handle-background-color: Specifies the background color of the switch handle.
+ * @prop --calcite-block-section-switch-handle-border-color-checked: Specifies the border color of the switch handle when checked.
  * @prop --calcite-block-section-switch-handle-border-color-hover: Specifies the border color of the switch handle on hover.
  * @prop --calcite-block-section-switch-handle-border-color: Specifies the border color of the switch handle.
```
[the remark this answers is not in the document]
Comment (alisonailea) [reply packages/calcite-components/src/components/block-section/block-section.scss:16]: Ah missed that thanks!

**Review by alisonailea — APPROVED** [review]

**Pull request merged**